File: src/main/java/org/gangel/graphomance/usecases/CreateBasicRelationTest.java
package org.gangel.graphomance.usecases;

import org.gangel.graphomance.NodeIdentifier;
import org.gangel.graphomance.RelationIdentifier;

import java.util.Map;

public class CreateBasicRelationTest extends TestBase {

    NodeIdentifier nodeId1;
    NodeIdentifier nodeId2;
    RelationIdentifier friendOf;

    @Override
    public void setUpTest() {
    }

    @Override
    public void createTestData() {
        schemaApi.createClass("Person");
        schemaApi.createRelationClass("FriendOf");
    }

    @Override
    public void performTest() {
        nodeId1 = objectApi.createNode("Person", Map.of("name", "John"));
        nodeId2 = objectApi.createNode("Person", Map.of("name", "Kate"));
        friendOf = objectApi.createRelation("FriendOf", nodeId1, nodeId2);
    }

    @Override
    public void cleanUpData() {
        objectApi.deleteAllRelations("FriendOf");
        objectApi.deleteAllNodes("Person");
        schemaApi.dropClass("Person");
        schemaApi.dropClass("FriendOf");
    }
}

File: src/main/kotlin/org/gangel/graphomance/usecases/CreateBasicRelationTest.kt
package org.gangel.graphomance.usecases

import org.gangel.graphomance.api.Session

class CreateBasicRelationTest : TestBase() {
	override fun setUpTest(session: Session) {}
	override fun createTestData(session: Session) {
		val schemaApi = session.schemaApi()
		schemaApi.createClass("Person")
		schemaApi.createRelationClass("FriendOf")
	}

	override fun performTest(session: Session) {
		val objectApi = session.objectApi()
		val nodeId1 = objectApi.createNode("Person", mapOf("name" to "John"))
		val nodeId2 = objectApi.createNode("Person", mapOf("name" to "Kate"))
		val friendOf = objectApi.createRelation("FriendOf", nodeId1, nodeId2)
	}

	override fun cleanUpData(session: Session) {
		val objectApi = session.objectApi()
		val schemaApi = session.schemaApi()
		objectApi.deleteAllRelations("FriendOf")
		objectApi.deleteAllNodes("Person")
		schemaApi.dropClass("Person")
		schemaApi.dropClass("FriendOf")
	}
}